### PR TITLE
Add WIFIonICE module

### PR DESCRIPTION
**Pull request by Kunsi**
This module adds support for displaying informations about the current trip on Deutsche Bahn high speed trains in i3pystatus. On click, travelynx (a popular trip-tracking tool) will open to allow for easy check in into the train.

TODOs:
- [x] Add documentation
- [x] Make output configurable to be able to show different parts of the information

### CODE REVIEW

**Comment by Kunsi** [comment]
This should now be ready to merge, will do a final test on wednesday when i'm on a train again

**Comment by Kunsi** [comment]
Is there anything else needed from my side to get this merged?

**Comment by terminalmage** [comment]
Sorry for the long delay... did you have any other work you wanted to do here?

**Comment by Kunsi** [comment]
No, this should be ready to merge.

File: i3pystatus/wifionice.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            minutes = int(seconds / 60)
+            seconds -= minutes * 60
+            components.append('{}m'.format(minutes))
+        if not components:
+            components.append('now')
+        return " ".join(components)
+
+    def _check_wifi(self):
+        if self.wifi_adapters is None:
+            self.logger.debug('self.wifi_adapters is None')
```
Comment (terminalmage) [inline i3pystatus/wifionice.py]: This log message could be improved. Maybe something like "automatic on-train detection is disabled"

File: i3pystatus/wifionice.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                    self.trip_info = loads(trip_info_req.read())['trip']
+
+                    ice_status_req = urlopen('https://iceportal.de/api1/rs/status')
+                    self.ice_status = loads(ice_status_req.read())
+                except Exception:
+                    self.trip_info = {}
+                    self.ice_status = {}
+
+                self.logger.debug(f'trip_info: {repr(self.trip_info)}')
+                self.logger.debug(f'ice_status: {repr(self.ice_status)}')
```
Comment (terminalmage) [inline i3pystatus/wifionice.py]: Python string formatting has built-in support for invoking `repr`, using `!r`. So, these could both be done using:

```python
self.logger.debug(f'trip_info: {self.trip_info!r}')
self.logger.debug(f'ice_status: {self.ice_status!r}')
```

File: i3pystatus/wifionice.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                with self.condition:
+                    self.condition.wait(self.on_train_interval)
+            else:
+                self.logger.info('Not on a train :(')
+
+                self.trip_info = {}
+                self.ice_status = {}
+
+                self.logger.debug(f'trip_info: {repr(self.trip_info)}')
+                self.logger.debug(f'ice_status: {repr(self.ice_status)}')
```
Comment (terminalmage) [inline i3pystatus/wifionice.py]: Same as above, regarding python string formatting.

File: i3pystatus/wifionice.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        if not components:
+            components.append('now')
+        return " ".join(components)
+
+    def _check_wifi(self):
+        if self.wifi_adapters is None:
+            self.logger.debug('self.wifi_adapters is None')
+            return True
+
+        from basiciw import iwinfo
```
Comment (terminalmage) [inline i3pystatus/wifionice.py]: With the default configuration, this import will cause a traceback unless basiciw is installed.

My personal preference is not to make the default configuration result in a traceback, and instead communicate a misconfiguration (or missing dependencies) via logging. However, a number of other i3pystatus modules result in a traceback if there are missing dependencies, and the default behavior of the `Status` class doesn't use a log file.

@enkore Do you have a specific preference for handling missing dependencies?

File: i3pystatus/wifionice.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+    def _format_time(self, seconds):
+        if seconds is None:
+            return "?"
+        seconds = int(seconds)
+        components = []
+        if seconds >= 3600:
+            hours = int(seconds / 3600)
+            seconds -= hours * 3600
+            components.append('{}h'.format(hours))
```
Comment (terminalmage) [inline i3pystatus/wifionice.py]: This can be an f-string

File: i3pystatus/wifionice.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+        if seconds >= 60:
+            minutes = int(seconds / 60)
+            seconds -= minutes * 60
+            components.append('{}m'.format(minutes))
```
Comment (terminalmage) [inline i3pystatus/wifionice.py]: This can also be an f-string

**Review by terminalmage — CHANGES_REQUESTED** [review]

File: i3pystatus/wifionice.py (hunk 4 above)
Comment (Kunsi) [reply i3pystatus/wifionice.py]: I have modified this to print a warning to the logs if basiciw is not installed.

**Comment by Kunsi** [comment]
Sorry it took me so long, but this should be good to go now :)

**Pull request merged**